src/sspi/ntlm.rs

```diff
@@ -257,6 +257,10 @@ impl SspiImpl for Ntlm {
             NtlmState::Authenticate => {
                 let input_token = SecurityBuffer::find_buffer(input, SecurityBufferType::Token)?;
 
+                if let Some(identity) = builder.credentials_handle {
+                    self.identity = identity.clone();
+                }
```

Comment:
Equivalent to:
```suggestion
                self.identity = identity.clone();
```

Reply:
It's not actually possible. `self.identity` is `Option<AuthIdentityBuffer>` and `builder.credentials_handle` is `Option<&mut Option<AuthIdentityBuffer>>`.
I changed it to `self.identity = builder.credentials_handle.and_then(|creds| creds.clone());`. I hope It's suitable for you.

Reply:
Oh okay, in that case what you need is the `.cloned()` method. Not a big deal anyway, I'll merge